extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -719,13 +721,162 @@ class SegmentationService {
 
     segDisplaySet.isLoaded = true;
 
-    this._broadcastEvent(EVENTS.SEGMENTATION_PIXEL_DATA_CREATED, {
+    this._broadcastEvent(EVENTS.SEGMENTATION_LOADING_COMPLETE, {
       segmentationId,
       segDisplaySet,
       overlappingSegments,
     });
 
-    return this.addOrUpdateSegmentation(segmentationSchema, suppressEvents);
+    return this.addOrUpdateSegmentation(segmentation, suppressEvents);
+  }
+
+  public async createSegmentationForRTDisplaySet(
+    rtDisplaySet,
+    segmentationId?: string,
+    suppressEvents = false
+  ): Promise<string> {
+    // Todo: we currently only have support for contour representation for initial
+    // RT display
+    const representationType = CONTOUR;
+    segmentationId = segmentationId ?? rtDisplaySet.displaySetInstanceUID;
+    const { structureSet } = rtDisplaySet;
+
+    if (!structureSet) {
+      throw new Error(
+        'To create the contours from RT displaySet, the displaySet should be loaded first, you can perform rtDisplaySet.load() before calling this method.'
+      );
+    }
+
+    const defaultScheme = this._getDefaultSegmentationScheme();
+    const rtDisplaySetUID = rtDisplaySet.displaySetInstanceUID;
+
+    const allRTStructData = structureSet.ROIContours.map(
```

Comment:
Can you extract a few of these functions out into separate files - something like `mapROIContoursToRTStructData` so that it is clear what that is doing.  It also means you could add a unit test on it, which might be useful to have.